From 17dfd77a94373f808646b1274d335adb8373604d Mon Sep 17 00:00:00 2001
From: Wenbo Zhu <b52619@freescale.com>
Date: Mon, 12 Oct 2015 10:16:03 +0800
Subject: [PATCH 541/602] MA-7094 when the board do cts test,
 com.android.cts.devicepolicy will fail(0910 daily
 sabresd sx)

remove feature android.software.managed_users from system.

Signed-off-by: Wenbo Zhu <b52619@freescale.com>
---
 imx6/evk_6sl.mk                   |  1 -
 imx6/sabreauto_6sx.mk             |  1 -
 imx6/sabresd_6sx.mk               |  1 -
 imx7/sabresd_7d.mk                |  1 -
 sabresd_6sx/required_hardware.xml | 42 +++++++++++++++++++++++++++++++++++++++
 5 files changed, 42 insertions(+), 4 deletions(-)

diff --git a/imx6/evk_6sl.mk b/imx6/evk_6sl.mk
index a47b613..55efaf7 100644
--- a/imx6/evk_6sl.mk
+++ b/imx6/evk_6sl.mk
@@ -48,7 +48,6 @@ PRODUCT_COPY_FILES += \
 	frameworks/native/data/etc/android.software.backup.xml:system/etc/permissions/android.software.backup.xml \
 	frameworks/native/data/etc/android.software.print.xml:system/etc/permissions/android.software.print.xml \
 	frameworks/native/data/etc/android.software.device_admin.xml:system/etc/permissions/android.software.device_admin.xml \
-	frameworks/native/data/etc/android.software.managed_users.xml:system/etc/permissions/android.software.managed_users.xml \
 	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
 	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
 	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
diff --git a/imx6/sabreauto_6sx.mk b/imx6/sabreauto_6sx.mk
index b28db2b..1cd7997 100644
--- a/imx6/sabreauto_6sx.mk
+++ b/imx6/sabreauto_6sx.mk
@@ -47,7 +47,6 @@ PRODUCT_COPY_FILES += \
 	frameworks/native/data/etc/android.software.backup.xml:system/etc/permissions/android.software.backup.xml \
 	frameworks/native/data/etc/android.software.print.xml:system/etc/permissions/android.software.print.xml \
 	frameworks/native/data/etc/android.software.device_admin.xml:system/etc/permissions/android.software.device_admin.xml \
-	frameworks/native/data/etc/android.software.managed_users.xml:system/etc/permissions/android.software.managed_users.xml \
 	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
 	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
 	frameworks/native/data/etc/android.hardware.sensor.light.xml:system/etc/permissions/android.hardware.sensor.light.xml \
diff --git a/imx6/sabresd_6sx.mk b/imx6/sabresd_6sx.mk
index 45944b6..05908f7 100644
--- a/imx6/sabresd_6sx.mk
+++ b/imx6/sabresd_6sx.mk
@@ -33,7 +33,6 @@ PRODUCT_CHARACTERISTICS := tablet
 PRODUCT_AAPT_CONFIG += xlarge large tvdpi hdpi
 
 PRODUCT_COPY_FILES += \
-	frameworks/native/data/etc/tablet_core_hardware.xml:system/etc/permissions/tablet_core_hardware.xml \
 	frameworks/native/data/etc/android.hardware.camera.xml:system/etc/permissions/android.hardware.camera.xml \
 	frameworks/native/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
 	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
diff --git a/imx7/sabresd_7d.mk b/imx7/sabresd_7d.mk
index c1c5f4f..56f39f5 100644
--- a/imx7/sabresd_7d.mk
+++ b/imx7/sabresd_7d.mk
@@ -48,7 +48,6 @@ PRODUCT_COPY_FILES += \
 	frameworks/native/data/etc/android.software.backup.xml:system/etc/permissions/android.software.backup.xml \
 	frameworks/native/data/etc/android.software.print.xml:system/etc/permissions/android.software.print.xml \
 	frameworks/native/data/etc/android.software.device_admin.xml:system/etc/permissions/android.software.device_admin.xml \
-	frameworks/native/data/etc/android.software.managed_users.xml:system/etc/permissions/android.software.managed_users.xml \
 	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
 	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
 	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
diff --git a/sabresd_6sx/required_hardware.xml b/sabresd_6sx/required_hardware.xml
index 0fd4fdf..05a61c8 100644
--- a/sabresd_6sx/required_hardware.xml
+++ b/sabresd_6sx/required_hardware.xml
@@ -19,4 +19,46 @@
 -->
 <permissions>
     <feature name="android.hardware.camera.any" />
+    <feature name="android.hardware.audio.output" />
+    <feature name="android.hardware.location" />
+    <feature name="android.hardware.location.network" />
+    <feature name="android.hardware.sensor.compass" />
+    <feature name="android.hardware.sensor.accelerometer" />
+    <feature name="android.hardware.bluetooth" />
+    <feature name="android.hardware.touchscreen" />
+    <feature name="android.hardware.touchscreen.multitouch" />
+    <feature name="android.hardware.touchscreen.multitouch.distinct" />
+    <feature name="android.hardware.microphone" />
+    <feature name="android.hardware.screen.portrait" />
+    <feature name="android.hardware.screen.landscape" />
+
+    <!-- basic system services -->
+    <feature name="android.software.app_widgets" />
+    <feature name="android.software.voice_recognizers" />
+    <feature name="android.software.backup" />
+    <feature name="android.software.home_screen" />
+    <feature name="android.software.input_methods" />
+    <feature name="android.software.print" />
+
+    <!-- Feature to specify if the device supports adding device admins. -->
+    <feature name="android.software.device_admin" />
+
+    <!-- devices with GPS must include android.hardware.location.gps.xml -->
+    <!-- devices with a rear-facing camera must include one of these as appropriate:
+         android.hardware.camera.xml or
+         android.hardware.camera.autofocus.xml or
+         android.hardware.camera.autofocus-flash.xml -->
+    <!-- devices with a front facing camera must include
+         android.hardware.camera.front.xml -->
+    <!-- devices with WiFi must also include android.hardware.wifi.xml -->
+    <!-- devices with an ambient light sensor must also include
+         android.hardware.sensor.light.xml -->
+    <!-- devices with a proximity sensor must also include
+         android.hardware.sensor.proximity.xml -->
+    <!-- devices with a barometer must also include
+         android.hardware.sensor.barometer.xml -->
+    <!-- devices with a gyroscope must also include
+         android.hardware.sensor.gyroscope.xml -->
+    <!-- GSM phones must also include android.hardware.telephony.gsm.xml -->
+    <!-- CDMA phones must also include android.hardware.telephony.cdma.xml -->
 </permissions>
-- 
1.8.0

